_posts/en/2019-10-21-schnorr-taproot-workshop.md

```diff
@@ -0,0 +1,189 @@
+name: 2019-10-21-schnorr-taproot-workshop
+type: posts
+layout: post
+lang: en
+slug: 2019-10-21-schnorr-taproot-workshop
+
+excerpt: >
+  A self-study course for learning about the schnorr/taproot softfork proposal.
+
+auto_id: false
+---
+{% include references.md %}
+
+In September 2019, Bitcoin Optech organized two workshops in San Francisco and
+New York on the schnorr/taproot softfork proposals. The aims of the workshop
+were to:
+
+1. share current thinking in the open source community on the proposals;
+2. give engineers a chance to work with the new technology through interactive
+   jupyter notebooks;
+3. help engineers take part in the community feedback process.
+
+This blog post contains all of the videos, slides and jupyter notebooks from
+those workshops, so engineers at home can learn about these exciting new
+technologies.
+
+The workshop is split into 4 sections:
+
+1. [Preparation and basic math](#preparation-and-basic-math) - shows how to set
+  up the jupyter notebook environment, gives a refresher on basic elliptic
+  curve math, and introduces tagged hashes.
+2. [Schnorr signatures and MuSig](#schnorr-signatures-and-musig) - describes
+  the bip-schnorr signature scheme, and how to use MuSig to aggregate multiple
+  public keys and partial signatures into a single pubkey/signature.
+3. [Taproot](#taproot) - explains how to embed a commitment to a single script
+  or multiple scripts into a tweaked public key. Shows how to spend a segwit v1
+  output using either the key path spend (with a single signature) or a script
+  path spend.
+4. [Case studies](#case-studies) - demonstrates some practical applications
+  of the new schnorr and taproot technologies.
+
+Slides from all the presentations can be downloaded [here][slides].
+Additionally, Bryan Bishop has provided a [transcript][] of the New York
+session.
+
+## Introduction
+
+[![Introduction](/img/posts/taproot-workshop/introduction.png)](https://www.youtube.com/watch?v=1gRCVLgkyAE&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC)
+{:.center}
+
+John Newbery introduces the workshop, giving a summary of why schorr and
+taproot are useful technologies. He also explains why Bitcoin Optech created
+the workshop, and what the aims of the workshop are.
+
+## Preparation and Basic Math
+
+This section shows how to setup the jupyter notebook, gives a refresher on
+basic elliptic curve math and introduces tagged hashes.
+
+#### 0.1 Test Notebook
+
+Before starting the workshops, users should follow the instructions in the
+repository [README][readme], clone the [workshop repository][] and run through
+the test notebook to ensure that their environment is set up correctly.
+
+#### 0.2 Elliptic Curve Math
+
+[![Introduction](/img/posts/taproot-workshop/elliptic-curve-math.png)](https://www.youtube.com/watch?v=oix8ov9iGgk&list=PLPrDsP88ifOVTEJf_jQGunDUS05M9GdIC&index=2)
+{:.center}
+
+Elichai Turkel gives a refresher on the basic elliptic curve math that will be
```

Comment:
The word "gives" is used repeatedly. Consider varying, for instance with "provides".